lib/urldata.h

```diff
@@ -60,13 +60,26 @@
  * protocol fields in the protocol handler.
  */
 #define CURLPROTO_WS     (1<<30)
-#define CURLPROTO_WSS    (1LL<<31)
-
-/* This type should be bumped to a curl_off_t once we need bit 32 or higher */
-typedef unsigned int curl_prot_t;
+#define CURLPROTO_WSS    ((curl_prot_t)1<<31)
 #else
 #define CURLPROTO_WS 0
 #define CURLPROTO_WSS 0
+#endif
+
+#define PROTO_TYPE_SMALL
+
+/* Protocols 32 and above must be added here as:
+ * #ifndef CURL_DISABLE_protoname
+ * #define CURLPROTO_protoname ((curl_prot_t) 1 << protonum)
+ * #undef PROTO_TYPE_SMALL
+ * #else
+ * #define CURLPROTO_protoname 0
+ * #endif
+ */
+
+#ifndef PROTO_TYPE_SMALL
+typedef curl_off_t curl_prot_t;
+#else
```

Comment:
This adds code and a comment *for the future* only, and I am generally against that and I am now too. Because most commonly, once that future arrives, some things have changed anyway causing things to get done differently.

Reply:
How would you then **prepare** the way for a developer to submit us a PR for a new protocol ?

Reply:
Adjust that commit to what is needed for the new protocol. I presume that would then remove the comment that says something about the future with actual code for the work in progress.

Reply:
Done